src/app/qgsoptions.cpp

```diff
@@ -1118,6 +1133,22 @@ void QgsOptions::saveOptions()
   }
   mSettings->setValue( QStringLiteral( "/browser/hiddenPaths" ), paths );
 
+  //QGIS help locations
+  myPaths.clear();
+  for ( int i = 0; i < mHelpPathTreeWidget->topLevelItemCount(); ++i )
+  {
+    QTreeWidgetItem* item = mHelpPathTreeWidget->topLevelItem( i );
+    if ( item )
+    {
+      if ( i != 0 )
+      {
+        myPaths += '|';
+      }
+      myPaths += item->text( 0 );
+    }
+  }
+  mSettings->setValue( QStringLiteral( "help/helpSearchPath" ), myPaths );
```

Comment:
Following the above comment, this would become:

    QStringList helpPaths;    
    for ( int i = 0; i < mHelpPathTreeWidget->topLevelItemCount(); ++i )
    {
      if ( QTreeWidgetItem* item = mHelpPathTreeWidget->topLevelItem( i ) )
      {
        helpPaths << item->text( 0 );
      }
    }
    mSettings->setValue( QStringLiteral( "help/helpSearchPath" ), helpPaths );